src/kaleidoscope/device/technomancy/Atreus.h

```diff
@@ -27,53 +27,44 @@
 #ifdef ARDUINO_AVR_ATREUS
 
 #include <Arduino.h>
-#define HARDWARE_IMPLEMENTATION kaleidoscope::hardware::technomancy::Atreus
-#include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
 
-#include "kaleidoscope/macro_helpers.h"
-
-#include "kaleidoscope/hardware/ATMegaKeyboard.h"
+#include "kaleidoscope/driver/keyscanner/AVR.h"
+#include "kaleidoscope/driver/bootloader/avr/HalfKay.h"
+#include "kaleidoscope/device/ATMega32U4Keyboard.h"
 
 namespace kaleidoscope {
-namespace hardware {
+namespace device {
 namespace technomancy {
-class Atreus: public kaleidoscope::hardware::ATMegaKeyboard {
-  friend class ATMegaKeyboard;
- public:
-  Atreus(void) {}
 
+struct AtreusProps : kaleidoscope::device::ATMega32U4KeyboardProps {
+  struct KeyScannerProps : public kaleidoscope::driver::keyscanner::AVRProps {
 #ifdef KALEIDOSCOPE_HARDWARE_ATREUS_PINOUT_ASTAR
-  ATMEGA_KEYBOARD_CONFIG(
-    ROW_PIN_LIST({PIN_D0, PIN_D1, PIN_D3, PIN_D2}),
-    COL_PIN_LIST({PIN_D7, PIN_C6, PIN_B5, PIN_B4, PIN_E6, PIN_D4, PIN_B6, PIN_F6, PIN_F7, PIN_D6, PIN_B7})
-  );
+    AVR_KEYSCANNER_PROPS(
```

Comment:
Do we want to move the ifdefs inside the AVR_KEYSCANNER_PROPS?

Reply:
Can do, yes. I was also toying with the idea of finding a way for the end-user to select a prop _without_ ifdefs, in their own sketch. That's why I didn't move them yet. But figuring this out is going to be a follow up PR if it happens, so I'll move the ifdefs.